datafusion/src/sql/planner.rs

```diff
@@ -1241,11 +1246,20 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         sql: &FunctionArg,
         schema: &DFSchema,
     ) -> Result<Expr> {
-        match sql {
-            FunctionArg::Named { name: _, arg } => {
-                self.sql_expr_to_logical_expr(arg, schema)
+        let arg: &FunctionArgExpr = match sql {
+            FunctionArg::Named { name: _, arg } => arg,
+            FunctionArg::Unnamed(arg) => arg,
+        };
+
+        match arg {
```

Comment:
Needed due to https://github.com/sqlparser-rs/sqlparser-rs/pull/378 from @panarch  (so we don't allow crazy stuff like `SELECT * + * from foo` 👍